cachito/common/checksum.py

```diff
@@ -1,34 +1,51 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+import base64
 import hashlib
+from pathlib import Path
+from typing import Union
+
 from cachito.errors import UnknownHashAlgorithm
 
 
-def get_hasher(algorithm: str = "sha256"):
-    """Get a hasher by a specific algorithm.
+class Hasher:
+    """A wrapper created from a hashlib hash object to provide custom digest methods."""
+
+    def __init__(self, hasher):
+        """Initialize this hasher object."""
+        self._hasher = hasher
+
+    def digest(self) -> bytes:
+        """Proxy of the hashlib hash object's digest method."""
+        return self._hasher.digest()
+
+    def hexdigest(self) -> str:
+        """Proxy of the hashlib hash object's hexdigest method."""
+        return self._hasher.hexdigest()
+
+    def base64digest(self) -> str:
+        """Return the digest encoded in base64."""
+        return base64.b64encode(self.digest()).decode("utf-8")
+
 
-    :param str algorithm: the algorithm name used to get a hasher from the hashlib module.
-    :return: the hasher got from the hashlib module.
+def hash_file(
+    file_path: Union[str, Path], chunk_size: int = 10240, algorithm: str = "sha256"
```

Comment:
Why `10240` ?

Reply:
Not sure. It was there originally. By goolging it quickly, seems the consideration behind this number is to have a good performance of reading files from the file system. But, I may be wrong to understand. :)

Reply:
ok, I was just curious